### PR TITLE
DAOS-15684 test: add test case for custom server name

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: pr,vm DaosSystemQuery
Skip-unit-tests: true
Skip-fault-injection-test: true

Add a case for changing the default name of "daos_server" Fix bug when setting config name

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/control/daos_system_query.yaml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 hosts:
   test_servers: 1
+  test_clients: 1
 timeout: 80
 server_config:
-  name: daos_server
+  name: custom_daos_server  # Use a non-default name
```
Comment (mchaarawi) [inline src/tests/ftest/control/daos_system_query.yaml:6]: i am not sure this test is a good candidate. i do not see it exercising the agent and client side stack to prove that this works.

Comment (daltonbohning) [reply src/tests/ftest/control/daos_system_query.yaml:6]: From a weekly run
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/weekly-testing/379/artifact/Functional%20on%20EL%208/control/daos_system_query.py/job.log

The test runs
- `dmg system query`
- `dmg storage format`
- `daos system query`

Do we need some real IO/app here for the agent+client? Or is `daos system query` enough?

Comment (mchaarawi) [reply src/tests/ftest/control/daos_system_query.yaml:6]: ah ok, daos_system_query should be good. i just did not see any agent_config in this yaml file ..

Comment (daltonbohning) [reply src/tests/ftest/control/daos_system_query.yaml:6]: The infrastructure will setup an agent config automatically :)

Comment (mchaarawi) [reply src/tests/ftest/control/daos_system_query.yaml:6]: this PR has been running for a long time. is it running all VM tests adn do they take that long?

Comment (daltonbohning) [reply src/tests/ftest/control/daos_system_query.yaml:6]: It's running `Test-tag: pr,vm DaosSystemQuery` which shouldn't take very long. It looks to me like the VM stage hasn't started? Or maybe Jenkins isn't reporting correctly

Comment (daltonbohning) [reply src/tests/ftest/control/daos_system_query.yaml:6]: I agree. Looks like PRs pushed after this one have gone through VM already. Let me try restarting

Comment (daltonbohning) [reply src/tests/ftest/control/daos_system_query.yaml:6]: Failed due to some CI issue. triggering again

**Comment by daltonbohning** [comment]
Interesting...


https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14225/3/artifact/Functional%20on%20EL%208.8/control/daos_system_query.py/job.log
```
  wolf-239vm2:
    -- Logs begin at Wed 2024-04-24 16:22:49 UTC, end at Wed 2024-04-24 17:31:30 UTC. --
    Apr 24 17:30:50 wolf-239vm2.wolf.hpdd.intel.com systemd[1]: Started DAOS Server.
    Apr 24 17:30:51 wolf-239vm2.wolf.hpdd.intel.com daos_server[117369]: ERROR: failed to load config from /etc/daos/daos_server.yml: invalid system name: "custom_daos_server"
```

The length limit is 15:
https://github.com/daos-stack/daos/blob/ddaf6ce295ab05b57abf60ca64b7452a415bd803/src/include/daos_types.h#L31

Although this exists?
https://github.com/daos-stack/daos/blob/ddaf6ce295ab05b57abf60ca64b7452a415bd803/src/include/daos_pool.h#L213
And is only used on one place:
https://github.com/daos-stack/daos/blob/ddaf6ce295ab05b57abf60ca64b7452a415bd803/src/client/dfs/duns.c#L1380

**Comment by mchaarawi** [comment]
> Interesting...
> 
> https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14225/3/artifact/Functional%20on%20EL%208.8/control/daos_system_query.py/job.log
> 
> ```
>   wolf-239vm2:
>     -- Logs begin at Wed 2024-04-24 16:22:49 UTC, end at Wed 2024-04-24 17:31:30 UTC. --
>     Apr 24 17:30:50 wolf-239vm2.wolf.hpdd.intel.com systemd[1]: Started DAOS Server.
>     Apr 24 17:30:51 wolf-239vm2.wolf.hpdd.intel.com daos_server[117369]: ERROR: failed to load config from /etc/daos/daos_server.yml: invalid system name: "custom_daos_server"
> ```
> 
> The length limit is 15:
> 
> https://github.com/daos-stack/daos/blob/ddaf6ce295ab05b57abf60ca64b7452a415bd803/src/include/daos_types.h#L31
> 
> Although this exists?
> 
> https://github.com/daos-stack/daos/blob/ddaf6ce295ab05b57abf60ca64b7452a415bd803/src/include/daos_pool.h#L213
> 
> 
> And is only used on one place:
> https://github.com/daos-stack/daos/blob/ddaf6ce295ab05b57abf60ca64b7452a415bd803/src/client/dfs/duns.c#L1380

yes not sure how we got into that situation. anyway made a small update to remove that duplicate def and use the control one.

**Comment by daltonbohning** [comment]
> yes not sure how we got into that situation. anyway made a small update to remove that duplicate def and use the control one.

Thanks :) At risk of setting  bad example, I update the pragmas to run unit tests and `pr`

**Review by mchaarawi — APPROVED** [review]

**Comment by daltonbohning** [comment]
Clean run:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-14225/6/pipeline

And the new test case shows it's working
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14225/6/artifact/Functional%20on%20EL%208.8/control/daos_system_query.py/job.log
```
2024-04-24 23:01:52,607 process          L0604 INFO | Running '/usr/bin/daos -j system query'
2024-04-24 23:01:52,844 process          L0416 DEBUG| [stdout] {
2024-04-24 23:01:52,845 process          L0416 DEBUG| [stdout]   "response": {
2024-04-24 23:01:52,846 process          L0416 DEBUG| [stdout]     "system_name": "other_dserver",
2024-04-24 23:01:52,846 process          L0416 DEBUG| [stdout]     "fabric_provider": "ofi+tcp",
2024-04-24 23:01:52,846 process          L0416 DEBUG| [stdout]     "rank_uris": [
2024-04-24 23:01:52,846 process          L0416 DEBUG| [stdout]       {
2024-04-24 23:01:52,846 process          L0416 DEBUG| [stdout]         "rank": 0,
2024-04-24 23:01:52,846 process          L0416 DEBUG| [stdout]         "uri": "ofi+tcp://10.8.4.23:31416"
2024-04-24 23:01:52,847 process          L0416 DEBUG| [stdout]       }
2024-04-24 23:01:52,847 process          L0416 DEBUG| [stdout]     ]
2024-04-24 23:01:52,847 process          L0416 DEBUG| [stdout]   },
2024-04-24 23:01:52,847 process          L0416 DEBUG| [stdout]   "error": null,
2024-04-24 23:01:52,847 process          L0416 DEBUG| [stdout]   "status": 0
2024-04-24 23:01:52,847 process          L0416 DEBUG| [stdout] }

```

**Review by phender — APPROVED** [review]

**Pull request merged**